padding/padding.go

```diff
@@ -0,0 +1,157 @@
+//Goal is to add padding to messages before encryption that will reduce
+//the amount of information leaked through the length of the message.
+//Specifically by O(loglog(l))
+
+//To pad a message with this all you need to do is call PadGeneric.
+//The first parameter is a byte slice with your message in it.
+//The second parameter is the number of bytes added by the encryption.
+//Currently this only works if the encryption adds a static overhead.
+
+package padding
+
+import (
+	"bytes"
+	"encoding/binary"
+	//	"fmt"
+	"math"
+)
+
+//overhead to encode the amount of padding,
+const PADLEN = 8
+
+//the byte used to pad messages.
+//0 because it makes things easier elsewhere, might need to be changed
```

Comment:
Since we are encrypting it anyways I don't think it needs to be random, although it is possible there are some benefits to doing random padding. Like protecting against use of bad symmetric/block ciphers that leak when encrypting the same byte over and over again. But I am not sure that really should be something we worry about.